src/buildercore/cfngen.py

```diff
@@ -418,14 +419,23 @@ def non_empty(self):
 _empty_cloudformation_dictionary = {'Resources': {}, 'Outputs': {}}
 Delta.__new__.__defaults__ = (_empty_cloudformation_dictionary, _empty_cloudformation_dictionary, _empty_cloudformation_dictionary, None)
 
+class TerraformDelta(namedtuple('TerraformDelta', ['old_contents', 'new_contents'])):
```

Comment:
I have never seen this before. Please add a comment or a link to just how this is working.

Reply:
`TerraformDelta`? Was just introduced here as a new class, adding a description.

Reply:
no - I mean, it looks like you're defining a class but inheriting from an instantiated object? It's as alien as: `class Foo({"bar?": "baz"}): ...` is to me.

Reply:
`namedtuple()` returns a class, that part seems standard:
https://docs.python.org/2/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields
Then I needed to add methods to that class, so I extended it with inheritance. No need for storing the original `namedtuple`-created class as only the extended one should be used. I'll add some comments about this pattern.